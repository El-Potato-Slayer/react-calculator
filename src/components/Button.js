import PropTypes from 'prop-types';

const Button = ({ name }) => (<div><p>{name}</p></div>);

Button.propTypes = {
  name: PropTypes.string,
};

Button.defaultProps = {
  name: '',
};

export default Button;
