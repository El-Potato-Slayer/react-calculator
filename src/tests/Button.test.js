import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import Button from '../components/Button';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('Renders Button component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button />, div);
});

it('Renders button correctly', () => {
  const { getByTestId } = render(<Button name="1" />);
  expect(getByTestId('button')).toHaveTextContent('1');
});

it('Renders button incorrectly', () => {
  const { getByTestId } = render(<Button name="1" />);
  expect(getByTestId('button')).not.toHaveTextContent('2');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Button name="1" />).toJSON();
  expect(tree).toMatchSnapshot();
});
