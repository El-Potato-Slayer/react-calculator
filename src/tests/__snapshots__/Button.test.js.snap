// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="button"
  data-testid="button"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
